lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java

```diff
@@ -0,0 +1,107 @@
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.lambda.powertools.logging.Logging;
+import software.amazon.lambda.powertools.tracing.Tracing;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.domain.JourneyResponse;
+import uk.gov.di.ipv.core.library.exceptions.HttpResponseExceptionWithErrorBody;
+import uk.gov.di.ipv.core.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+import uk.gov.di.ipv.core.library.helpers.RequestHelper;
+import uk.gov.di.ipv.core.library.service.ConfigurationService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+import uk.gov.di.ipv.core.library.service.UserIdentityService;
+
+import java.util.List;
+
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.ADDRESS_CRI_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.FRAUD_CRI_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.KBV_CRI_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.PASSPORT_CRI_ID;
+
+public class SelectCriHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final String CRI_START_JOURNEY = "/journey/%s";
+    public static final String JOURNEY_FAIL = "/journey/fail";
+
+    private final ConfigurationService configurationService;
+    private final UserIdentityService userIdentityService;
+    private final IpvSessionService ipvSessionService;
+
+    public SelectCriHandler(
+            ConfigurationService configurationService,
+            UserIdentityService userIdentityService,
+            IpvSessionService ipvSessionService) {
+        this.configurationService = configurationService;
+        this.userIdentityService = userIdentityService;
+        this.ipvSessionService = ipvSessionService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public SelectCriHandler() {
+        this.configurationService = new ConfigurationService();
+        this.userIdentityService = new UserIdentityService(configurationService);
+        this.ipvSessionService = new IpvSessionService(configurationService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent event, Context context) {
+        LogHelper.attachComponentIdToLogs();
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(event);
+            String userId = ipvSessionService.getUserId(ipvSessionId);
```

Comment:
Depending on which PR goes in first, be good here to attach the new govukSigninJourneyId we're adding in #391

Reply:
Added this in https://github.com/alphagov/di-ipv-core-back/pull/399